Lib/datetime.py

```diff
@@ -1226,9 +1293,9 @@ class time:
     Properties (readonly):
     hour, minute, second, microsecond, tzinfo, fold
     """
-    __slots__ = '_hour', '_minute', '_second', '_microsecond', '_tzinfo', '_hashcode', '_fold'
+    __slots__ = '_hour', '_minute', '_second', '_microsecond', '_submicrosecond', '_tzinfo', '_hashcode', '_fold'
 
-    def __new__(cls, hour=0, minute=0, second=0, microsecond=0, tzinfo=None, *, fold=0):
+    def __new__(cls, hour=0, minute=0, second=0, microsecond=0, tzinfo=None, *, fold=0, submicrosecond=0):
```

Comment:
Ugh. I don't like `submicrosecond` being keyword-only. I also don't like breaking compatibility with code that passes all arguments through `tzinfo` positionally. Anyone want to reassure me this isn't awful?

Reply:
Yeah, It is. We can change later by deprecating passing tzinfo as positionally

Reply:
What's the rationale behind `submicrosecond` over `nanosecond`? What if at some point picosecond accuracy is desired?